catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/KmlInputTransformer.java

```diff
@@ -0,0 +1,101 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.kml.transformer;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.platform.util.TemporaryFileBackedOutputStream;
+import org.codice.ddf.spatial.kml.converter.KmlToMetacard;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.CharStreams;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import de.micromata.opengis.kml.v_2_2_0.Kml;
+
+public class KmlInputTransformer implements InputTransformer {
+    private MetacardType metacardType;
+
+    public KmlInputTransformer(MetacardType metacardType) {
+        this.metacardType = metacardType;
+    }
+
+    @Override
+    public Metacard transform(InputStream inputStream)
+            throws IOException, CatalogTransformerException {
+        return transform(inputStream, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream inputStream, String id)
+            throws IOException, CatalogTransformerException {
+
+        MetacardImpl metacard;
+
+        try (TemporaryFileBackedOutputStream fileBackedOutputStream = new TemporaryFileBackedOutputStream()) {
+
+            try {
+                IOUtils.copy(inputStream, fileBackedOutputStream);
+
+            } catch (IOException e) {
+                throw new CatalogTransformerException(
+                        "Unable to transform KML to Metacard. Error reading input stream.",
+                        e);
+            } finally {
+                IOUtils.closeQuietly(inputStream);
+            }
+
+            try (InputStream inputStreamCopy = fileBackedOutputStream.asByteSource()
+                    .openStream()) {
+                metacard = (MetacardImpl) unmarshal(inputStreamCopy);
+            }
+
+            if (metacard == null) {
+                throw new CatalogTransformerException("Unable to transform Kml to Metacard.");
+            } else if (StringUtils.isNotEmpty(id)) {
+                metacard.setAttribute(Metacard.ID, id);
+            }
+
+            String kmlString;
+            try (Reader reader = fileBackedOutputStream.asByteSource()
+                    .asCharSource(Charsets.UTF_8)
+                    .openStream()) {
+                kmlString = CharStreams.toString(reader);
+            }
+
+            metacard.setAttribute(Metacard.METADATA, kmlString);
```

Comment:
Should we move this up into the try?  I think that way we can only set it if it doesn't throw an IOException

Reply:
Yep. Moved